rllib/algorithms/sac/sac_rl_module.py

```diff
@@ -0,0 +1,106 @@
+from typing import Type
+from ray.rllib.algorithms.sac.sac_catalog import SACCatalog
+
+# from ray.rllib.algorithms.sac.sac_learner import QF_PREDS
+# from ray.rllib.core.models.base import ACTOR, CRITIC
+from ray.rllib.core.models.specs.typing import SpecType
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.core.rl_module.rl_module_with_target_networks_interface import (
+    RLModuleWithTargetNetworksInterface,
+)
+from ray.rllib.models.distributions import Distribution
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import ExperimentalAPI, override
+
+CRITIC_TARGET = "critic_target"
+QF_PREDS = "qf_preds"
+
+
+@ExperimentalAPI
+class SACRLModule(RLModule, RLModuleWithTargetNetworksInterface):
+    def setup(self):
+        # __sphinx_doc_begin__
+        # super().setup()
+        catalog: SACCatalog = self.config.get_catalog()
+
+        # Build the encoder for the policy.
+        self.pi_encoder = catalog.build_encoder(framework=self.framework)
+
+        # SAC needs next to the pi network also a Q network with encoder.
+        # Note, as the Q network uses next to the observations, actions as
+        # input we need an additional encoder for this.
+        self.qf_encoder = catalog.build_qf_encoder(framework=self.framework)
+
+        # Build the target Q encoder as an exact copy of the Q encoder.
+        # TODO (simon): Maybe merging encoders together for target and qf
+        # and keep only the heads differently?
+        self.qf_target_encoder = catalog.build_qf_encoder(framework=self.framework)
+
+        # Build heads.
+        self.pi = catalog.build_pi_head(framework=self.framework)
+        self.qf = catalog.build_qf_head(framework=self.framework)
+        # The Q target network head is an identical copy of the Q network head.
+        self.qf_target = catalog.build_qf_head(framework=self.framework)
```

Comment:
Dumb question: What about the twin networks? We don't support this currently? Would it be hard to add?